### PR TITLE
Inductor cpp wrapper: change inputs args from tuple to vector

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89743
* #90755
* #89742
* #89744
* __->__ #90754
* #88667
* #88666
* #88561

## Pitch
Change input args type from `std::tuple` to `std::vector` to reduce the compilation time.

## Description
`std::tie()` takes quite a long time during the compilation when the input args number grows.

For example, for a graph from the `PegasusForConditionalGeneration` model with 318 input args, the compilation of `std::tie` for the args is about 10s. By changing to std::vector, the compilation time of arg assignment is reduced to less than 1s.

### Code before:
```cpp
at::Tensor call_0(std::tuple<at::Tensor&, at::Tensor&> args) {
    at::Tensor arg0_1, arg1_1;
    std::tie(arg0_1, arg1_1) = args;
    ...
    return buf0; 
}
```

### Code after:
```cpp
at::Tensor call_0(std::vector<at::Tensor> args) {
    at::Tensor arg0_1, arg1_1;
    arg0_1 = args[0];
    arg1_1 = args[1];
    ...
    return buf0; 
}
```

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/codegen/wrapper.py [hunk 1]
```diff
@@ -613,16 +613,15 @@ def write_prefix(self):
...
+                    f"{output_types} call_{self._call_func_id}({inputs_types} args) {{"
                 )
                 inputs_keys_str = ", ".join(V.graph.graph_inputs.keys())
                 self.wrapper_call.writeline(f"at::Tensor {inputs_keys_str};")
-                self.wrapper_call.writeline(f"std::tie({inputs_keys_str}) = args;")
+                for idx, input_key in enumerate(V.graph.graph_inputs.keys()):
+                    self.wrapper_call.writeline(f"{input_key} = args[{idx}];")
             else:
                 self.wrapper_call.writeline(
                     f"{output_types} call_{self._call_func_id}(std::tuple<> args) {{"
```
Comment (jgong5) [inline torch/_inductor/codegen/wrapper.py:627]: How about we always do `std::vector` now? For no arg case, it could be an empty vector.

**Review by jgong5 — APPROVED** [review]
I'm OK with this temporary solution while we should eventually do CPython binding and take care of object life cycles and manage the memory footprint well.

File: torch/_inductor/codegen/wrapper.py (hunk 1 above)
Comment (chunyuan-w) [reply torch/_inductor/codegen/wrapper.py]: I've simplified the code to use `std::vector` for both `inputs_len >= 1` and `inputs_len == 0`.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**